src/hammer-vlsi/hammer_vlsi/cli_driver.py

```diff
@@ -7,23 +7,35 @@
 #  See LICENSE for licence details.
 
 import argparse
+import importlib.util
 import json
 import os
 import subprocess
 import sys
+import tempfile
+import warnings
 
+if importlib.util.find_spec("ruamel.yaml") is None:
```

Comment:
1) I think it might need to be `importlib.util.find_spec("ruamel")` for it not to trigger the error? Either way, it seems like `ruamel` should be an optional dependency, so the CI should still work even if ruamel is not present. I will look at the CI again afterwards.

2) Since this line is repeated everywhere I suggest adding it as a util function and just calling it as `if has_ruamel():` etc.

Reply:
Sounds good, will implement the changes.